x-pack/plugins/security_solution/public/explore/components/paginated_table/helpers.ts

```diff
@@ -20,3 +20,30 @@ export const generateTablePaginationOptions = (
     querySize: isBucketSort ? limit : limit + cursorStart,
   };
 };
+
+export const getLimitedPaginationOptions = (
+  activePage: number,
+  limit: number
+): PaginationInputPaginatedInput => {
+  const cursorStart = activePage * limit;
+  return {
+    activePage,
+    cursorStart,
+    querySize: limit + cursorStart,
+    fakePossibleCount: 0, // TODO: remove this
+  };
+};
+
+export const getLimitedPaginationTotalCount = ({
+  activePage,
+  limit,
+  totalCount,
+}: {
+  activePage: number;
+  limit: number;
+  totalCount: number;
+}): number => {
+  const fakePossibleCount =
+    4 <= activePage && activePage > 0 ? limit * (activePage + 2) : limit * 5;
```

Comment:
Couple of ideas:

1. When reading `4 <= activePage` my brain first inverts it to `activePage >= 4` and then analyzes it. Similarly to how `!notSomething` my brain struggles to first convert it to positive `Something` and then analyze. I think I read somewhere about this phenomenon of how our brain works, so there  is some chance I am not the only one who feels like that. But up to you of course.

2. `(activePage + 2)` and `(limit * 5)` look pretty magical to me, can you add a comment or better put them in expressive variables to explain what they mean?

Reply:
I agree. This is old code I moved around, I'll revisit this logic to make it easier to understand.